planner/core/logical_plan_builder.go

```diff
@@ -701,19 +701,14 @@ func (b *PlanBuilder) buildProjection4Union(u *LogicalUnionAll) {
 	// Process each child and add a projection above original child.
 	// So the schema of `UnionAll` can be the same with its children's.
 	for childID, child := range u.children {
-		uniqueIDToOutputIdxes := make(map[int][]int)
-		if p, ok := child.(*LogicalProjection); ok {
-			for i, expr := range p.Exprs {
-				col, ok1 := expr.(*expression.Column)
-				if !ok1 {
-					continue
-				}
-				uniqueID := int(col.UniqueID)
-				uniqueIDToOutputIdxes[uniqueID] = append(uniqueIDToOutputIdxes[uniqueID], i)
-			}
-		}
+		childProj, isChildProj := child.(*LogicalProjection)
 		exprs := make([]expression.Expression, len(child.Schema().Columns))
 		for i, srcCol := range child.Schema().Columns {
+			if isChildProj {
+				if exprCol, isCol := childProj.Exprs[i].(*expression.Column); isCol{
```

Comment:
We'd better avoid to directly write column's `UniqueID`.
I prefer original way.
Also, I'll test other ways to solve this.